libs/base/di/src/injector.ts

```diff
@@ -127,21 +135,27 @@ export class Injector {
     }
 
     if (providerInstance.instance === undefined) {
-      const restoreInjector = setCurrentInjector(this);
-
-      if (isClassProvider(providerInstance.provider)) {
+      if (isAsyncClassProvider(providerInstance.provider)) {
+        providerInstance.instance = createLazyProxy(
+          providerInstance.provider.asyncClass,
+          this,
+          (instance) => (providerInstance.instance = instance)
+        );
+      } else if (isClassProvider(providerInstance.provider)) {
+        const restoreInjector = setCurrentInjector(this);
         providerInstance.instance = new providerInstance.provider.useClass();
+        restoreInjector();
```

Comment:
It looks like we are changing a logic of the method a bit. Are we considering it as a fix or a breaking change?

Reply:
we need to restore it, only when setCurrentInjector was called first (when we instantiate service), and in case of asnycClass, we don't need it (as we delay instantation). So the logic for other cases (where it's needed) behaves the same.